### PR TITLE
Add the documentation for MapboxNavigationControl &  MapboxPopup components

### PR DESCRIPTION
**Pull request by perruche**
Add documentation for undocumented components 

✅  MapboxNavigationControl
❌ MapboxGeocoder
❌ MapboxSource
✅   MapboxPopup
❌ MapboxLayer

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -55,6 +55,8 @@
     "eslint-plugin-vue": "^5.0.0",
     "husky": "^3.0.0",
     "mapbox-gl": "^1.1.0",
+    "node-sass": "^4.13.0",
+    "sass-loader": "^8.0.0",
```
Comment (perruche) [inline package.json]: Is there a way to exclude those dependencies from the package that is publish to npm @titouanmathis ? It's only useful for the documentation

File: docs/components/MapboxPopup.md [hunk 2]
```diff
@@ -6,4 +6,90 @@ next: ./MapboxSource.md
...
+### `maxWidth`
+
+- Type `[ String ]`
+- Required `false`
+- default `240px`
+
+Set a max-width to the popup.
+
+## Events
+@todo
```
Comment (titouanmathis) [inline docs/components/MapboxPopup.md]: There are only 2 events: `open` and `close`, you can find their documentation here: https://docs.mapbox.com/mapbox-gl-js/api/#popup.event:open

File: docs/components/MapboxPopup.md [hunk 3]
```diff
@@ -6,4 +6,90 @@ next: ./MapboxSource.md
 
 # MapboxPopup
 
-...
+Display a popup on the map.
+
+- [Mapbox popup example](https://docs.mapbox.com/mapbox-gl-js/example/popup/)
```
Comment (titouanmathis) [inline docs/components/MapboxPopup.md]: We should have a link to the `Popup` documentation here : https://docs.mapbox.com/mapbox-gl-js/api/#popup

File: docs/components/MapboxPopup.md [hunk 4]
```diff
@@ -6,4 +6,90 @@ next: ./MapboxSource.md
...
+- default `true`
+
+Close the popup on close button click.
+
+### `anchor`
+
+- Type `[ String ]`
+- Required `false`
+
+@todo.
```
Comment (titouanmathis) [inline docs/components/MapboxPopup.md]: > A string indicating the part of the Popup that should be positioned closest to its coordinate. See https://docs.mapbox.com/mapbox-gl-js/api/#popup for more details.

File: docs/components/MapboxPopup.md [hunk 5]
```diff
@@ -6,4 +6,90 @@ next: ./MapboxSource.md
...
+### Basic usage
+
+<client-only>
+  <mapbox-popup-demo api-key="MAPBOX_API_KEY" />
+</client-only>
+
+<<< @/docs/.vuepress/components/MapboxPopupDemo.vue
+
+### Advance usage
+@todo
```
Comment (titouanmathis) [inline docs/components/MapboxPopup.md]: We can remove the "Advanced usage" section for now. We will add one when needed 😉

File: docs/components/MapboxNavigationControl.md [hunk 6]
```diff
@@ -6,4 +6,47 @@ next: ./MapboxPopup.md
 
 # MapboxNavigationControl
 
-...
+Display navigation controls on the map, including two zoom buttons + - and a compass button to rotate the map.
+
+- [Mapbox navigation controls example](https://docs.mapbox.com/mapbox-gl-js/example/navigation/)
+
```
Comment (titouanmathis) [inline docs/components/MapboxNavigationControl.md]: We should have a link to the official documentation here: https://docs.mapbox.com/mapbox-gl-js/api/#navigationcontrol 🙂

File: docs/components/MapboxPopup.md (hunk 4 above)
Comment (perruche) [reply docs/components/MapboxPopup.md]: Thanks i was not entierly sure of what the prop was doing, and haven't got time to dig the doc

File: docs/components/MapboxPopup.md (hunk 2 above)
Comment (perruche) [reply docs/components/MapboxPopup.md]: i update the events section, i let you look into it

**Pull request merged**